torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -995,10 +995,28 @@ TensorView* sub_alpha(TensorView* v1, TensorView* v2, Val* v3) {
 }
 // lerp
 Val* lerp(Val* start, Val* end, Val* weight) {
+  auto casted_values =
+      promoteValues(TypePromotion::default_op_config, {start, end, weight});
+  start = casted_values[0];
+  end = casted_values[1];
+  weight = casted_values[2];
+
+  auto out_dtype =
```

Comment:
lerp in eager mode does not promote types

Reply:
promote_type just returns the higher precision type between the two. If PyTorch lerp doesn't support mixed type, this function does nothing and we'll have the same behavior from a parsed graph.